state/azure/cosmosdb/cosmosdb.go

```diff
@@ -414,16 +418,15 @@ func (c *StateStore) Set(ctx context.Context, req *state.SetRequest) error {
 	pk := azcosmos.NewPartitionKeyString(partitionKey)
 	_, err = c.client.UpsertItem(upsertCtx, pk, marsh, &options)
 	if err != nil {
+		resErr := &azcore.ResponseError{}
```

Comment:
Looks like Cosmos DB was not reporting Etag failures as separate errors in Set operations. In a follow-up PR I'll update our conformance tests to make sure that all components report the correct errors.

Reply:
The same changes was made for other components too.

Follow-up PR is #2843